### ISSUE TITLE
error when use CylindricalSurface?

### ISSUE THREAD

**Issue by lyyc199586** [fzaverl/s3dlib#1]
Curretnly s3dlib uses matplotlib 3.0.2, current version of matplotlib is 3.8.1, and that seems to raise some conflicts when I try to run the tutorial code in README.

```
---------------------------------------------------------------------------
AttributeError                            Traceback (most recent call last)
[c:\workspace\cm-vis\cm_vis\scheme\shapes_3d.py](file:///C:/workspace/cm-vis/cm_vis/scheme/shapes_3d.py) in line 15
     [18](file:///c%3A/workspace/cm-vis/cm_vis/scheme/shapes_3d.py?line=17)     return R, 2*t, Z
     [20](file:///c%3A/workspace/cm-vis/cm_vis/scheme/shapes_3d.py?line=19) # 2. Setup and map surface  .........................................
---> [21](file:///c%3A/workspace/cm-vis/cm_vis/scheme/shapes_3d.py?line=20) surface = s3d.CylindricalSurface(6)
     [22](file:///c%3A/workspace/cm-vis/cm_vis/scheme/shapes_3d.py?line=21) surface.map_cmap_from_op(lambda c: c[1],'hsv')
     [23](file:///c%3A/workspace/cm-vis/cm_vis/scheme/shapes_3d.py?line=22) surface.map_geom_from_op( knot )

File [~\AppData\Local\Packages\PythonSoftwareFoundation.Python.3.10_qbz5n2kfra8p0\LocalCache\local-packages\Python310\site-packages\s3dlib\surface.py:4585](https://file+.vscode-resource.vscode-cdn.net/c%3A/workspace/cm-vis/~/AppData/Local/Packages/PythonSoftwareFoundation.Python.3.10_qbz5n2kfra8p0/LocalCache/local-packages/Python310/site-packages/s3dlib/surface.py:4585), in CylindricalSurface.__init__(self, rez, basetype, name, **kwargs)
ref='~\AppData\Local\Packages\PythonSoftwareFoundation.Python.3.10_qbz5n2kfra8p0\LocalCache\local-packages\Python310\site-packages\s3dlib\surface.py:0'>0</a>;32m   [4582](file:///c%3A/Users/liuya/AppData/Local/Packages/PythonSoftwareFoundation.Python.3.10_qbz5n2kfra8p0/LocalCache/local-packages/Python310/site-packages/s3dlib/surface.py?line=4581) indexObj,vertexCoor = self.triangulateBase(rez,baseVcoor,baseFaceVertexIndices, self._midVectorFun)
ref='~\AppData\Local\Packages\PythonSoftwareFoundation.Python.3.10_qbz5n2kfra8p0\LocalCache\local-packages\Python310\site-packages\s3dlib\surface.py:0'>0</a>;32m   [4583](file:///c%3A/Users/liuya/AppData/Local/Packages/PythonSoftwareFoundation.Python.3.10_qbz5n2kfra8p0/LocalCache/local-packages/Python310/site-packages/s3dlib/surface.py?line=4582) faceIndices = indexObj['face']
ref='~\AppData\Local\Packages\PythonSoftwareFoundation.Python.3.10_qbz5n2kfra8p0\LocalCache\local-packages\Python310\site-packages\s3dlib\surface.py:1'>1</a>;32m-> [4585](file:///c%3A/Users/liuya/AppData/Local/Packages/PythonSoftwareFoundation.Python.3.10_qbz5n2kfra8p0/LocalCache/local-packages/Python310/site-packages/s3dlib/surface.py?line=4584) super().__init__(vertexCoor, faceIndices, name, **kwargs)
ref='~\AppData\Local\Packages\PythonSoftwareFoundation.Python.3.10_qbz5n2kfra8p0\LocalCache\local-packages\Python310\site-packages\s3dlib\surface.py:0'>0</a>;32m   [4587](file:///c%3A/Users/liuya/AppData/Local/Packages/PythonSoftwareFoundation.Python.3.10_qbz5n2kfra8p0/LocalCache/local-packages/Python310/site-packages/s3dlib/surface.py?line=4586) self.coorType = _COORSYS["CYLINDRICAL"]
   [4588](file:///c%3A/Users/liuya/AppData/Local/Packages/PythonSoftwareFoundation.Python.3.10_qbz5n2kfra8p0/LocalCache/local-packages/Python310/site-packages/s3dlib/surface.py?line=4587) self._normal_scale =  _normLength( len(faceIndices), [1](https://file+.vscode-resource.vscode-cdn.net/c%3A/workspace/cm-vis/~/AppData/Local/Packages/PythonSoftwareFoundation.Python.3.10_qbz5n2kfra8p0/LocalCache/local-packages/Python310/site-packages/s3dlib/surface.py:1).14, 1.31)

File [~\AppData\Local\Packages\PythonSoftwareFoundation.Python.3.10_qbz5n2kfra8p0\LocalCache\local-packages\Python310\site-packages\s3dlib\surface.py:649](https://file+.vscode-resource.vscode-cdn.net/c%3A/workspace/cm-vis/~/AppData/Local/Packages/PythonSoftwareFoundation.Python.3.10_qbz5n2kfra8p0/LocalCache/local-packages/Python310/site-packages/s3dlib/surface.py:649), in Surface3DCollection.__init__(self, vertexCoor, faceIndices, name, **kwargs)
ref='~\AppData\Local\Packages\PythonSoftwareFoundation.Python.3.10_qbz5n2kfra8p0\LocalCache\local-packages\Python310\site-packages\s3dlib\surface.py:0'>0</a>;32m    [647](file:///c%3A/Users/liuya/AppData/Local/Packages/PythonSoftwareFoundation.Python.3.10_qbz5n2kfra8p0/LocalCache/local-packages/Python310/site-packages/s3dlib/surface.py?line=646) # ========================================================================
ref='~\AppData\Local\Packages\PythonSoftwareFoundation.Python.3.10_qbz5n2kfra8p0\LocalCache\local-packages\Python310\site-packages\s3dlib\surface.py:0'>0</a>;32m    [648](file:///c%3A/Users/liuya/AppData/Local/Packages/PythonSoftwareFoundation.Python.3.10_qbz5n2kfra8p0/LocalCache/local-packages/Python310/site-packages/s3dlib/surface.py?line=647) self.coorType = _COORSYS["XYZ"]
--> [649](file:///c%3A/Users/liuya/AppData/Local/Packages/PythonSoftwareFoundation.Python.3.10_qbz5n2kfra8p0/LocalCache/local-packages/Python310/site-packages/s3dlib/surface.py?line=648) self.baseSurfaceColor = np.array(self._facecolors3d[[0](https://file+.vscode-resource.vscode-cdn.net/c%3A/workspace/cm-vis/~/AppData/Local/Packages/PythonSoftwareFoundation.Python.3.10_qbz5n2kfra8p0/LocalCache/local-packages/Python310/site-packages/s3dlib/surface.py:0)]).flatten().tolist()
    [651](file:///c%3A/Users/liuya/AppData/Local/Packages/PythonSoftwareFoundation.Python.3.10_qbz5n2kfra8p0/LocalCache/local-packages/Python310/site-packages/s3dlib/surface.py?line=650) self.vertexColor = np.array(self._facecolors3d[[0](https://file+.vscode-resource.vscode-cdn.net/c%3A/workspace/cm-vis/~/AppData/Local/Packages/PythonSoftwareFoundation.Python.3.10_qbz5n2kfra8p0/LocalCache/local-packages/Python310/site-packages/s3dlib/surface.py:0)]).copy()[np.newaxis,:]
ref='~\AppData\Local\Packages\PythonSoftwareFoundation.Python.3.10_qbz5n2kfra8p0\LocalCache\local-packages\Python310\site-packages\s3dlib\surface.py:0'>0</a>;32m    [652](file:///c%3A/Users/liuya/AppData/Local/Packages/PythonSoftwareFoundation.Python.3.10_qbz5n2kfra8p0/LocalCache/local-packages/Python310/site-packages/s3dlib/surface.py?line=651) # following assigned when surface color applied using an operation

AttributeError: 'CylindricalSurface' object has no attribute '_facecolors3d'
```

**Comment by zifengdexiatian** [comment]
When I was using version 1.1.0, I encountered the same error, but after switching to the older version with 'pip install s3dlib==1.0.0', it was able to run properly

**Comment by lyyc199586** [comment]
closed as update to 1.2.0 fixed it
